alternatives.md

```diff
@@ -0,0 +1,113 @@
+
+<!-- TODO -->
+
+## [ctags](http://ctags.sourceforge.net/)
+
+> Ctags generates an index (or tag) file of language objects found in source files that allows these items to be quickly and easily located by a text editor or other utility. A tag signifies a language object for which an index entry is available (or, alternatively, the index entry created for that object).
+
+Both Babelfish and Ctags provides positional information for identifiers,
+classes, directives, etc.
+
+The main difference is that Ctags does not provide any form of AST, while
+Babelfish provides native language AST as well as language-independent UAST.
+
+Also, Babelfish ecosystem allows making more complex queries over AST.
+
+## [ANTLR](https://github.com/antlr/antlr4)
+
+> ANTLR (ANother Tool for Language Recognition) is a powerful parser generator for reading, processing, executing, or translating structured text or binary files. It's widely used to build languages, tools, and frameworks. From a grammar, ANTLR generates a parser that can build parse trees and also generates a listener interface (or visitor) that makes it easy to respond to the recognition of phrases of interest.
+
+ANTLR is a language generator with support for most existing languages
+and data formats. But parsers usually produce parse trees that are usually
+different from native language AST, thus requiring additional processing
+to be used for analysis. The structure of parse tree is also language-dependent.
+
+Babelfish provides a correct native language AST as well as
+language-independent UAST for all supported languages.
+
+## [Tree-sitter](https://github.com/tree-sitter/tree-sitter)
+
+> Tree-sitter is a C library for incremental parsing, intended to be used via bindings to higher-level languages. It can be used to build a concrete syntax tree for a program and efficiently update the syntax tree as the program is edited. This makes it suitable for use in text-editing programs.
+
+Tree-sitter provides a simplified AST to be able to execute a limited set
+of queries. It usually cannot provide enough details suitable for static
+analysis. Instead, it focuses on performance and implements real-time
+tree diffing.
+
+Babelfish provides a full native language AST as well as language-independent
+UAST for all supported languages. It is suitable for static analysis since
+it preserves all features of original AST.
+
+## [srcML](https://www.srcml.org/)
+
+> The srcML format is an XML representation for source code, where the markup tags identify elements of the abstract syntax for the language. The srcml program is a command line application for the conversion source code to srcML, an interface for the exploration, analysis, and manipulation of source code in this form, and the conversion of srcML back to source code. The current parsing technologies supports C/C++, C#, and Java.
```

Comment:
support

Reply:
The fun thing is that it's the quote from their site.